android/src/main/java/io/ably/flutter/plugin/push/FirebaseMessagingReceiver.java

```diff
@@ -28,8 +29,15 @@
 
   @Override
   public void onReceive(Context context, Intent intent) {
+    if (intent.getExtras() == null){
+      return;
+    }
+
     listenForFlutterApplicationToFinishProcessingMessage(context);
-    sendMessageToFlutterApplication(context, intent);
+
+    Executors.newSingleThreadExecutor().execute(() -> {
```

Comment:
Isn't it too much memory consuming to create a new thread executor for each push notification we receive? :thinking:

Reply:
You are right. I provided a single static executor service to be reused here https://github.com/ably/ably-flutter/pull/291/commits/07a865cdbcbf229e89cc283b7ca0a095b85e0480.